tools/mc-java/src/main/java/io/spine/tools/mc/java/codegen/MessagesConfig.java

```diff
@@ -0,0 +1,148 @@
+                           .setProperty(String.class)
+                           .convention(ImmutableList.of());
+        nestedClassFactories = p.getObjects()
+                                .setProperty(String.class)
+                                .convention(ImmutableList.of());
+    }
+
+    void emptyByConvention() {
+        interfaceNames().convention(ImmutableSet.of());
+        methodFactories.convention(ImmutableSet.of());
+        nestedClassFactories.convention(ImmutableSet.of());
+    }
+
+    /**
+     * Specifies a {@link io.spine.tools.protoc.MethodFactory} to generate methods for
+     * the message classes.
+     *
+     * <p>Calling this method multiple times will add the provided factories for code generation.
+     *
+     * @param factoryClassName
+     *         the canonical class name of the method factory
+     */
+    public void generateMethodsWith(String factoryClassName) {
+        methodFactories.add(factoryClassName);
+    }
+
+    /**
+     * Specifies a {@link io.spine.tools.protoc.NestedClassFactory} to generate nested classes
+     * inside the message classes.
+     *
+     * <p>Calling this method multiple times will add the provided factories for code generation.
+     *
+     * @param factoryClassName
+     *         the canonical class name of the nested class factory
+     */
+    public void generateNestedClassesWith(String factoryClassName) {
+        nestedClassFactories.add(factoryClassName);
+    }
+
+    @Override
+    Messages toProto() {
+        Messages.Builder result = Messages.newBuilder()
+                .setPattern(pattern)
+                .addAllAddInterface(interfaces())
+                .addAllGenerateMethods(generateMethods())
+                .addAllGenerateNestedClasses(generateNestedClasses());
+        GenerateFields generateFields = generateFields();
+        if (isNotDefault(generateFields)) {
+            result.setGenerateFields(generateFields);
+        }
+        return result.build();
+    }
+
+    private Set<GenerateMethods> generateMethods() {
+        return methodFactories.get()
+                              .stream()
+                              .map(MessagesConfig::methodFactoryConfig)
+                              .collect(toSet());
+    }
+
+    private static GenerateMethods methodFactoryConfig(String methodFactoryClass) {
+        MethodFactoryName factoryName = MethodFactoryName.newBuilder()
+                .setClassName(className(methodFactoryClass))
+                .build();
+        GenerateMethods config = GenerateMethods.newBuilder()
+                .setFactory(factoryName)
+                .build();
+        return config;
+    }
+
+    private Set<GenerateNestedClasses> generateNestedClasses() {
+        return nestedClassFactories.get()
+                                   .stream()
+                                   .map(MessagesConfig::nestedClassFactoryConfig)
+                                   .collect(toSet());
```

Comment:
Can we extend `SetProperty` so that we can do the thing `get().stream().map().collect()` in the derived class once and then use it all over the places just passing the arguments of the `map()` call?